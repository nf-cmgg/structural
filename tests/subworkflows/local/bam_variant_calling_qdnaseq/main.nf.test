nextflow_workflow {

    name "Test Workflow BAM_VARIANT_CALLING_QDNASEQ"
    script "subworkflows/local/bam_variant_calling_qdnaseq/main.nf"
    workflow "BAM_VARIANT_CALLING_QDNASEQ"

    tag "subworkflows"
    tag "subworkflows_local"
    tag "bam_variant_calling_qdnaseq"

    config "./nextflow.config"

    test("homo_sapiens") {

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [id:"test", sample:"test", sex:"male"],
                    file(params.cram1, checkIfExists: true),
                    file(params.crai1, checkIfExists: true)
                ], [
                    [id:"test2", sample:"test2", sex:"female"],
                    file(params.cram2, checkIfExists: true),
                    file(params.crai2, checkIfExists: true)
                ])
                input[1] = Channel.value([
                    [id:"fasta"],
                    file(params.fasta, checkIfExists: true)
                ])
                input[2] = Channel.value([
                    [id:"fai"],
                    file(params.fai, checkIfExists: true)
                ])
                input[3] = Channel.value([
                    [id:"male_ref"],
                    file(params.qdnaseq_male, checkIfExists: true)
                ])
                input[4] = Channel.value([
                    [id:"female_ref"],
                    file(params.qdnaseq_female, checkIfExists: true)
                ])
                input[5] = Channel.fromPath("\${params.bedgovcf_configs}/*.yaml", checkIfExists:true).collect()
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.vcf,
                    workflow.out.qdnaseq_beds,
                    workflow.trace.succeeded().size(),
                    workflow.trace.failed().size()
                ).match() }
            )
        }

    }

}
