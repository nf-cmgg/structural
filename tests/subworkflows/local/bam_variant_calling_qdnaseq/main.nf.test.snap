{
    "homo_sapiens": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "sample": "test",
                        "sex": "male"
                    },
                    "test.qdnaseq.vcf.gz:md5,d032c063dd19057fea0f9c985f845a34"
                ],
                [
                    {
                        "id": "test2",
                        "sample": "test2",
                        "sex": "female"
                    },
                    "test2.qdnaseq.vcf.gz:md5,d4d7b8884cc1ba0147482f724a4a40b7"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "sample": "test",
                        "sex": "male",
                        "caller": "qdnaseq"
                    },
                    "test.bed:md5,19be72c3f56297070c6070ded3419dee"
                ],
                [
                    {
                        "id": "test2",
                        "sample": "test2",
                        "sex": "female",
                        "caller": "qdnaseq"
                    },
                    "test2.bed:md5,fed14c95db6bed194f40f262d68399d1"
                ]
            ],
            10,
            0
        ],
        "timestamp": "2023-11-29T16:39:34.961850643"
    }
}